ykrt/src/compile/jitc_yk/jit_ir.rs

```diff
@@ -646,9 +738,66 @@ impl Module {
                 return ConstIdx::new(idx);
             }
         }
-        // type table miss, we need to insert it.
+        // const table miss, we need to insert it.
         self.push_const(c.clone())
     }
+
+    /// Get the index of a type, inserting it into the type table if necessary.
+    pub(crate) fn type_idx(&mut self, t: &Type) -> Result<TypeIdx, CompilationError> {
+        // FIXME: can we optimise this?
+        for (idx, tt) in self.types.iter_enumerated() {
+            if tt == t {
+                // type table hit.
+                return Ok(idx);
+            }
+        }
+        // type table miss, we need to insert it.
+        self.push_type(t.clone())
+    }
+
+    /// Get the index of a function declaration, inserting it into the func decl table if necessary.
+    pub(crate) fn func_decl_idx(&mut self, d: &FuncDecl) -> Result<FuncDeclIdx, CompilationError> {
+        // FIXME: can we optimise this?
+        for (idx, td) in self.func_decls.iter_mut_enumerated() {
+            if td == d {
+                // func decl table hit.
+                return Ok(idx);
+            }
+        }
+        // type table miss, we need to insert it.
+        self.push_func_decl(d.clone())
+    }
+
+    /// Get the index of a global, inserting it into the global declaration table if necessary.
+    pub(crate) fn global_idx(&mut self, g: &Global) -> Result<GlobalIdx, CompilationError> {
+        // FIXME: can we optimise this?
+        for (idx, tg) in self.globals.iter_mut_enumerated() {
+            if tg == g {
+                // global decl table hit.
+                return Ok(idx);
+            }
+        }
+        // global decl table miss, we need to insert it.
+        self.push_global(g.clone())
+    }
+
+    /// Return the [FuncType] of the specified function declaration.
+    ///
+    /// # Panics
+    ///
+    /// Panics if:
+    ///  - `idx` is out of bounds
+    ///  - the type index inside the [FuncDecl] s out of bounds
```

Comment:
s/s/is

Reply:
:snake:

Reply:
Fixed in https://github.com/ykjit/yk/pull/982/commits/fff51cb5167ecbb01dbfddf30b409e5c4294d032

:badger: